pkg/traceql/test_examples.yaml

```diff
@@ -0,0 +1,229 @@
+# valid queries parse successfully and return nil when calling .validate()
+valid:
+  # spanset filters
+  - '{ true }'
+  - '{ !true }'
+  - '{ true && false }'
+  - '{ true || false }'
+  - '{ 1 = 2 }'
+  - '{ 1 != 2 }'
+  - '{ 1 > 2 }'
+  - '{ 1 >= 2 }'
+  - '{ 1 < 2 }'
+  - '{ 1 <= 2 }'
+  - '{ 1 + 1 = 2 }'
+  - '{ 1 - 1 = 2 }'
+  - '{ 1 * 1 = 2 }'
+  - '{ 1 / 1 = 2 }'
+  - '{ 1 ^ 1 = 2 }'
+  - '{ -1 = 2 }'
+  - '{ "test" =~ "test" }'
+  - '{ "test" !~ "test" }'
+  - '{ "test" = "test" }'
+  - '{ "test" != "test" }'
+  - '{ .a }'
+  - '{ !.a }'
+  - '{ .a && false }'
+  - '{ .a || true }'
+  - '{ .a = 2 }'
+  - '{ .a != 2 }'
+  - '{ .a > 2 }'
+  - '{ .a >= 2 }'
+  - '{ .a < 2 }'
+  - '{ .a <= 2 }'
+  - '{ .a + 1 = 2 }'
+  - '{ .a - 1 = 2 }'
+  - '{ .a * 1 = 2 }'
+  - '{ .a / 1 = 2 }'
+  - '{ .a ^ 1 = 2 }'
+  - '{ -.a = 2 }'
+  - '{ .a =~ "test" }'
+  - '{ .a !~ "test" }'
+  - '{ .a = "test" }'
+  - '{ .a != "test" }'
+  - '{ parent.a != 3 }'
+  - '{ parent.resource.a && true }'
+  - '{ parent.span.a > 3 }'
+  - '{ parent.duration = 1h }'
+  - '{ resource.a != 3 }'
+  - '{ span.a != 3 }'
+  - '{ !("test" != .c || ((true && .b) || 3 < .a)) }'
+  - '{ parent = nil }'
+  - '{ status = ok }'
+  - '{ status = unset }'
+  - '{ status = error }'
+  - '{ status != error }'
+  - '{ duration > 1s }'
+  - '{ duration > 1s * 2s }' 
```

Comment:
My thought was just to default to go which would i believe multiply `2s * 1s` as nanoseconds which give a completely nonsensical answer. The reason I figured we'd go this way is due to the complexity of resolving whether or not this particular combination of floats, ints, durations and operators makes sense.

My guess is that it would be better to be open and let the use type nonsensical things then be strict and miss some cases that made sense? Perhaps it's not as hard to I think to come up with all correct and incorrect cases. Currently all combinations parse and validate.

Some examples that make sense:
```
1s + 2s
1s = 2s
1s / 2
1s * 2
```

Some examples that don't make sense:
```
1s * 2s
1s = 2
```